site/gatsby-site/src/components/submissions/SubmissionList.js

```diff
@@ -175,8 +214,139 @@ const SubmissionList = ({ data }) => {
     );
   }
 
+  const [selectedAction, setSelectedAction] = useState('claim');
+
+  const [performingAction, setPerformingAction] = useState(false);
+
+  const bulkActions = async () => {
+    try {
+      const selectedSubmissions = Object.keys(selectedRows).filter(
+        (sr) => selectedRows[sr] === true
+      );
+
+      if (selectedSubmissions.length > 0) {
+        if (selectedAction === 'claim') {
```

Comment:
Could you please refactor this code to extract three different functions for each action (claim, unclaim, and reject)?

Reply:
I refactored it with one function since they were very similar and there was a lot of repeated code. Let me know if you think it's a good and understandable solution